### PR TITLE
lint: Add linter to detect wrong file permissions

### PR DESCRIPTION
**Pull request by cvengler**
This adds a test to avoid stuff like #17806

### CODE REVIEW

**Comment by emilengler** [comment]
Interesting, the linter says a specific file has the wrong file permission but on my system it has the correct already

File: test/lint/check-permissions.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# This test checks if all test files have the 755 unix permission (-rwxr-xr-x)
+
+import os
+import stat
+from os import listdir
+from pathlib import Path
```
Comment (fjahr) [inline test/lint/check-permissions.py]: unused

File: test/lint/check-permissions.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# This test checks if all test files have the 755 unix permission (-rwxr-xr-x)
+
+import os
+import stat
```
Comment (fjahr) [inline test/lint/check-permissions.py]: unused

**Review by fjahr — APPROVED** [review]
Concept ACK

But currently, another linter fails on your script, see my comments :)

Edit: commit message should probably be changed to "lint: Add linter to detect wrong file permissions"

**Comment by jonatack** [comment]
-0.1 on concept, -0.8 on execution. Has this ever come up with other contributions in the past? Without further justification this seems to be addressing a non-issue with code that needs to be maintained.

**Comment by emilengler** [comment]
>  Has this ever come up with other contributions in the past? 

Yes see #17806

**Comment by emilengler** [comment]
Looks like this is impossible with because of git.
It only supports two filemodes

**Comment by MarcoFalke** [comment]
This check could be a one line check in the test_runner.py before the script is executed?

**Pull request closed without merging**